app/controllers/foreman_register/hosts_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module ForemanRegister
+  class HostsController < ::ForemanRegister::ApplicationController
+    # Skip default filters
+    FILTERS = [
+      :require_login,
+      :session_expiry,
+      :update_activity_time,
+      :set_taxonomy,
+      :authorize,
+      :verify_authenticity_token
+    ].freeze
+
+    FILTERS.each do |f|
```

Comment:
We could extract this to a concern, `Foreman::Controller::AnonymousAccess` or something similar.